### ISSUE TITLE
"Greater than" component erroneously reports 'true'

### ISSUE THREAD

**Issue by koraximus** [Regalis11/Barotrauma#7909]
- [ X] I have searched the issue tracker to check if the issue has already been reported.

**Description**
I have a wiring arrangement wherein a greater than component always reports a True condition, even when I rearrange the inputs.

The WiFi component receives signal from another Wifi on the same channel, into which an air sensor outputs air%.
![Screenshot (167)](https://user-images.githubusercontent.com/96561533/147192426-e71d71c2-9c14-40ab-a166-d283bce4a30b.png)

**Steps To Reproduce**
For the first input, I have wired a memory component outputting 55. The second input comprises a WiFi component that outputs to the Greater than via a relay.

[MEM]-->55-->[Greater]<--[Relay]<--[WiFi_out]<-----\\----- [WiFi_in]<---Air%

I have confirmed the output of the greater than component by using display screens, and they show that it always outputs a 1 unless change the true output to something else.

**Version**
v0.15.22.1  Windows.

**Additional information**
I also tried 'inverting' the output, so that it sent a zero when true and 1 when false, but it still only reported true (so it started sending a zero). Interestingly, this remained true even when a nearby text display showed that the other input (Wifi out/Relay) dipped below the mem--> 55 input, further indicating that the Greater component is simply locked into a true state.

I have attached a screenshot showing the wiring of the greater than component in question.

**Comment by Tsunder** [comment]
perhaps the GT is not outputting any signal at all?  :)

the text display only updates when it gets a new value.

use a terminal instead of a text display

**Comment by MaaiG** [comment]
https://github.com/Regalis11/Barotrauma/issues/7844#issuecomment-997204018

**Comment by Regalis11** [comment]
I gave this a test as well (with a similar circuit as the one in MaaiG's clip), and couldn't spot any issues. @koraximus if the problem isn't that the greater than component's false output isn't set, could you send the submarine file or an item assembly containing the circuit here so I can give it a test and see if I can diagnose the issue?

**Comment by koraximus** [comment]
Regalis11,

I recreated this circuit in a fresh file and the problem resolved. I still cannot pinpoint the problem in my original file but it is clear that this is not a persistent issue and it is most likely that I overlooked some small detail somewhere. I recommend that this issue be closed.

Edit: I think the issue may have been that the Greater than component received a null signal by default. I have observed in unrelated testing that logic with a null input does not align with my intuitions.
